src/main/java/pl/javastart/task/Person.java

```diff
@@ -0,0 +1,61 @@
+package pl.javastart.task;
+
+public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private String pesel;
+
+    public Person(String firstName, String lastName, int age, String pesel) {
+        setFirstName(firstName);
+        setLastName(lastName);
+        setAge(age);
+        this.pesel = pesel;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        if (firstName == null || firstName.length() < 2) {
+            throw new NameUndefinedException();
```

Comment:
Jeżeli chcesz rzucać wyjątek z własnym komunikatem, tak jak napisałas w komentarzu do pull requesta, czyli `throw new NameUndefinedException("Imię jest nieprawidłowe!" + firstName);`, to musisz w klasie NameUndefinedException stworzyć sobie konstruktor z argumentem String, a następnie zapisać ten komunikat przez wywołanie super().
Przykład w tej lekcji:
https://app.javastart.pl/kurs/java/java-podstawy-temat-23-wyjatki/lekcja/java-wlasne-wyjatki-new